py/mkrules.mk

```diff
@@ -102,15 +102,19 @@ $(BUILD)/frozen.c: $(wildcard $(FROZEN_DIR)/*) $(HEADER_BUILD) $(FROZEN_EXTRA_DE
 endif
 
 ifneq ($(FROZEN_MPY_DIR),)
+# to build the MicroPython cross compiler
+$(TOP)/mpy-cross/mpy-cross: $(TOP)/py/*.[ch] $(TOP)/mpy-cross/*.[ch] $(TOP)/windows/fmode.c
```

Comment:
Personally, I think I'd be inclined to just drop the dependenies and make $(TOP)/mpy-cross/mpy-cross be declared .PHONY so that it always runs make. Make should run pretty fast (less than a second on my machine) if it's up-to-date.

so something like:
```
.PHONY: $(TOP)/mpy-cross/mpy-cross
$(TOP)/mpy-cross/mpy-cross:
    $(Q)$(MAKE) -C $@
```
Then you don't run into any weird things where you forgot some dependency.